modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/ActiveTip.scala

```diff
@@ -0,0 +1,10 @@
+package org.tessellation.dag.snapshot
+
+import org.tessellation.dag.domain.block.BlockReference
+
+import eu.timepit.refined.types.numeric.NonNegLong
+
+case class ActiveTip(
+  blockReference: BlockReference,
+  usageCount: NonNegLong
```

Comment:
I wonder if we should have the count as a number vs a `Set` of child blocks hashes. For sure we would need to have the additional field for artificial usages and the total usage would be `childBlocks.size() + artificialUsages`. The advantage of a `Set` is that adding the hash again would be idempotent, with counts you need to be sure that the given block didn't increase the count already.

Reply:
We had similar issue on the current mainnet - we were not able to distinguish why tip has been used (because of which parent), so we've started using `Set` with children hashes.

Reply:
Likely not needed. Two reasons:
* We will be able to tell if the child is attempted to be accepted more than once, based on the tips (each block is a tip, and parents are not removed before children)
* We need to have 'artificial' usages on top of regular usages.